ios/l3_interfaces/ios_l3_interface.yaml

```diff
@@ -0,0 +1,102 @@
+---
+GENERATOR_VERSION: '1.0'
+
+ANSIBLE_METADATA: |
+    {
+      'metadata_version': '1.1',
+      'status': ['preview'],
+      'supported_by': 'network'
+    }
+NETWORK_OS: ios
+RESOURCE: l3_interfaces
+COPYRIGHT: Copyright 2019 Red Hat
+
+DOCUMENTATION: |
+  module: ios_l3_interfaces
+  version_added: 2.9
+  short_description: Manage Layer-3 interface on Cisco IOS devices.
+  description: This module provides declarative management of Layer-3 interface on Cisco IOS devices.
+  author: Sumit Jaiswal (@justjais)
+  options:
+    config:
+      description: A dictionary of Layer-3 interface options
+      type: list
+      elements: dict
+      suboptions:
+        name:
+          description:
+          - Full name of the interface excluding any logical unit number, i.e. GigabitEthernet0/1.
+          type: str
+          required: True
+        ipv4:
+          description:
+          - IPv4 address to be set for the Layer-3 interface mentioned in I(name) option.
+            The address format is <ipv4 address>/<mask>, the mask is number in range 0-32
+            eg. 192.168.0.1/24
+          suboptions:
+            address:
+              description:
+              - Configures the IPv4 address for Interface.
+              type: str
+            secondary:
```

Comment:
@cidrblock have included support for `secondary` IP and also support for handling ipv4 `dhcp` and ipv6 `autoconfig`.